build/images/ovs/Dockerfile.windows

```diff
@@ -14,34 +14,37 @@
 
 ARG NANOSERVER_VERSION
 
-FROM mcr.microsoft.com/windows/servercore:${NANOSERVER_VERSION} as windows-ovs
+FROM mcr.microsoft.com/windows/servercore:${NANOSERVER_VERSION} as ovs-base
 
 ARG WIN_OVS_VERSION
 
-# Switch to a privileged user
-USER ContainerAdministrator
-
 # Sets the shell to be PowerShell 
 SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 
 ENV WIN_OVS_VERSION ${WIN_OVS_VERSION}
 
 # Install VC for ovs
 RUN curl.exe -Lo C:\vc_redist.x64.exe https://aka.ms/vs/17/release/vc_redist.x64.exe; \
-    Start-Process -FilePath C:\vc_redist.x64.exe -ArgumentList ‘/install’,’/quiet’,’/norestart’ -Verb RunAs -Wait; \
-    rm C:\vc_redist.x64.exe
+    Start-Process -FilePath C:\vc_redist.x64.exe -ArgumentList ‘/install’,’/quiet’,’/norestart’ -Verb RunAs -Wait;
 
 # Download and extract OVS
 RUN curl.exe -fLo C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip https://downloads.antrea.io/ovs/ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip; \
-    Expand-Archive C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip -DestinationPath C:\  ; \
-    rm C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip
+    Expand-Archive C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip -DestinationPath C:\;
 
 # Download, extract OpenSSL and copy libraries
 RUN curl.exe -Lo C:\openssl-1.0.2u-x64_86-win64.zip https://indy.fulgan.com/SSL/openssl-1.0.2u-x64_86-win64.zip; \
-    Expand-Archive C:\openssl-1.0.2u-x64_86-win64.zip -DestinationPath C:\openssl  ;  \
-    Move-Item -Path 'C:\openssl\libeay32.dll', 'C:\openssl\ssleay32.dll' -Destination 'C:\Windows\System32\' ; \
-    Remove-Item -Path 'C:\openssl-1.0.2u-x64_86-win64.zip' ;\
-    Remove-Item -Path 'C:\openssl' -Recurse -Force
+    Expand-Archive C:\openssl-1.0.2u-x64_86-win64.zip -DestinationPath C:\openssl;
 
+FROM mcr.microsoft.com/powershell:lts-nanoserver-${NANOSERVER_VERSION} as windows-ovs
 SHELL ["cmd", "/S", "/C"]
+
+# Switch to a privileged user
+USER ContainerAdministrator
+
+COPY --from=ovs-base /Windows/System32/vcruntime140.dll /Windows/System32/
+COPY --from=ovs-base /openssl/libeay32.dll /Windows/System32/
+COPY --from=ovs-base /openssl/ssleay32.dll /Windows/System32/
+
+COPY --from=ovs-base /openvswitch /openvswitch/
+
 RUN setx /m PATH "C:/openvswitch/usr/bin;C:/openvswitch/usr/sbin;%PATH%"
```

Comment:
Is this line necessary ? (Since we run as hostprocess=true ) Please see https://github.com/antrea-io/antrea/pull/4936#discussion_r1254913894

Reply:
I leave the command here inside the antrea-ovs image is to ensure we can use this image to run OVS commands separately even if it is not integrated with antrea-windows Pod. As antrea-agent-windows Pod is using image antrea/antrea-windows which is different from antrea/windows-ovs, our settings in this Dockerfile does not work for the Pod containers. Your mentioning lines are to ensure antrea-agent Pod running as expect.